### PR TITLE
Simplify curvilinear grid examples.

### PR DESCRIPTION
**Pull request by anntzer**
Since 8840d6e we ensure that `x` and `y` passed to `tr` and `inv_tr`
are already arrays (see the definition of
`_UserTransform2D.transform_non_affine`), so we don't need additional
casting-to-arrays.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: examples/axisartist/demo_curvelinear_grid.py [hunk 1]
```diff
@@ -27,13 +27,8 @@ def curvelinear_test1(fig):
...
 
-    def tr(x, y):
-        x, y = np.asarray(x), np.asarray(y)
-        return x, y - x
-
-    def inv_tr(x, y):
-        x, y = np.asarray(x), np.asarray(y)
-        return x, y + x
+    def tr(x, y): return x, y - x
+    def inv_tr(x, y): return x, y + x
```
Comment (timhoffm) [inline examples/axisartist/demo_curvelinear_grid.py:31]: Style: I don't think we should put function bodies on the same line (and I'm surprised flake8 doesn't complain). Either give it proper line breaks (preferred), or inline them into `GridHelperCurveLinear using lambdas.

Comment (anntzer) [reply examples/axisartist/demo_curvelinear_grid.py:31]: One liners can be written together, see https://www.python.org/dev/peps/pep-0008/#blank-lines / https://github.com/PyCQA/pycodestyle/pull/823.  I can still spread things out if you prefer, just let me know.

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Test failure is real:
```
Traceback (most recent call last):
  File "/home/circleci/project/examples/axisartist/demo_curvelinear_grid.py", line 112, in <module>
    curvelinear_test1(fig)
  File "/home/circleci/project/examples/axisartist/demo_curvelinear_grid.py", line 41, in curvelinear_test1
    xx, yy = tr([3, 6], [5, 10])
  File "/home/circleci/project/examples/axisartist/demo_curvelinear_grid.py", line 30, in tr
    def tr(x, y): return x, y - x
TypeError: unsupported operand type(s) for -: 'list' and 'list
```

**Review by timhoffm — CHANGES_REQUESTED** [review]
Test fails.

**Comment by anntzer** [comment]
Indeed, fixed.

**Review by timhoffm — APPROVED** [review]

**Pull request merged**